compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1546,7 +1552,7 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
       }
     case tp1: TypeVar if tp1.isInstantiated =>
       tp1.underlying & tp2
-    case tp1: AnnotatedType =>
+    case tp1: AnnotatedType if !tp1.isRefining =>
       tp1.underlying & tp2
```

Comment:
Why not rewrap the annotation around the intersection if it's a refining annotation?